qt/FreeSpaceLabel.cc

```diff
@@ -73,33 +66,35 @@ FreeSpaceLabel::onTimer ()
   tr_variantInitDict (&args, 1);
   tr_variantDictAddStr (&args, TR_KEY_path, myPath.toUtf8 ().constData());
 
-  myTag = mySession->getUniqueTag ();
-  mySession->exec ("free-space", &args, myTag);
-}
-
-void
-FreeSpaceLabel::onSessionExecuted (int64_t tag, const QString& result, tr_variant * arguments)
-{
-  Q_UNUSED (result);
+  RpcQueue * q = new RpcQueue;
 
-  if (tag != myTag)
-    return;
-
-  QString str;
-
-  // update the label
-  int64_t bytes = -1;
-  if (tr_variantDictFindInt (arguments, TR_KEY_size_bytes, &bytes) && bytes >= 0)
-    setText (tr("%1 free").arg(Formatter::sizeToString (bytes)));
-  else
-    setText (QString ());
-
-  // update the tooltip
-  size_t len = 0;
-  const char * path = 0;
-  tr_variantDictFindStr (arguments, TR_KEY_path, &path, &len);
-  str = QString::fromUtf8 (path, len);
-  setToolTip (QDir::toNativeSeparators (str));
+  q->addReq (
+    [this, &args] (RpcResponse)
```

Comment:
OK, I'll finally refactor RpcQueue to accept all combinations of "(don't) have error handler" + "(don't) have return value" + "(don't) have argument" in an overhead-less manner... Hope you don't mind an additional twice explicitly specialized template helper class.